anitya/admin.py

```diff
@@ -375,3 +375,116 @@ def browse_logs():
         from_date=from_date or '',
         user=user or ''
     )
+
+
+@APP.route('/flags')
+@login_required
+def browse_flags():
+
+    if not is_admin():
+        flask.abort(401)
+
+    from_date = flask.request.args.get('from_date', None)
+    state = flask.request.args.get('state', None)
+    project = flask.request.args.get('project', None)
+    user = flask.request.args.get('user', None)
+    refresh = flask.request.args.get('refresh', False)
+    limit = flask.request.args.get('limit', 50)
+    page = flask.request.args.get('page', 1)
+
+    try:
+        page = int(page)
+    except ValueError:
+        page = 1
+
+    try:
+        int(limit)
+    except ValueError:
+        limit = 50
+        flask.flash('Incorrect limit provided, using default', 'errors')
+
+    if from_date:
+        try:
+            from_date = parser.parse(from_date)
+        except (ValueError, TypeError):
+            flask.flash(
+                'Incorrect from_date provided, using default', 'errors')
+            from_date = None
+
+    if from_date:
+        from_date = from_date.date()
+
+    offset = 0
+    if page is not None and limit is not None and limit != 0:
+        offset = (page - 1) * limit
+
+    flags = []
+
+    try:
+        flags = anitya.lib.model.ProjectFlag.search(
+            SESSION,
+            project_name=project or None,
+            state=state or None,
+            from_date=from_date,
+            user=user or None,
+            offset=offset,
+            limit=limit,
+        )
+
+        cnt_flags = anitya.lib.model.ProjectFlag.search(
+            SESSION,
+            project_name=project or None,
+            state=state or None,
+            from_date=from_date,
+            user=user or None,
+            count=True
+        )
+    except Exception, err:
+        import logging
+        logging.exception(err)
+        flask.flash(err, 'errors')
+
+    total_page = int(ceil(cnt_flags / float(limit)))
+
+    return flask.render_template(
+        'flags.html',
+        current='flags',
+        refresh=refresh,
+        flags=flags,
+        cnt_flags=cnt_flags,
+        total_page=total_page,
+        page=page,
+        project=project or '',
+        from_date=from_date or '',
+        user=user or ''
+    )
+
+
+@APP.route('/flags/<flag_id>/set/<state>', methods=['GET'])
```

Comment:
OK, I'm still a bit confused.
- Do I need to wrap a `<form>` around the individual button, or the entire `for` loop in the template that renders X buttons?
- How do I set the form `action=` if there are many buttons that can have one of two actions? I assume this means I need to wrap a `<form>` around each button, answering my first question?
- Which part of that `delete_distro` function actually adds CSRF protection? `validate_on_submit`?
- If I need to use a form for these buttons to avoid a security hole, should I just use the already-existing (empty) `ConfirmationForm`, or something else?